### PR TITLE
Added private and public filters in playlist plugin

### PR DESCRIPTION
**Pull request by AmbrishRamachandiran**
## Hey, I just made a Pull Request!

Added private and public filters 

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kuangp** [comment]
Thanks for this but I'm not sure placing this filter inside the "Personal" section is the best UI since public and private are not "personal filters" - what do you think about creating a separate dropdown "Visibility" filter ?

**Review by tudi2d — CHANGES_REQUESTED** [review]
Before diving into the code: Could you please provide more information on why you want to make this update and what your proposal entails? It might be helpful to create a "[Feature" issue](https://github.com/backstage/backstage/issues/new/choose) first, enabling others to give feedback & than create a PR later on. This way the risks of having to rewrite your code or doing work, which is not approved is much lower as the initial idea was validated before. Additionally it would allow bundling several smaller changes to one plugin into one PR (saw[ your documentation change form the other day](https://github.com/backstage/backstage/pull/20326)).

TL;DR: Suggestion for now is to close this PR & bring this discussion to a Feature-issue first.

**Comment by AmbrishRamachandiran** [comment]
@tudi2d  @kuangp  @Rugvip closing is created the issues for suggestions https://github.com/backstage/backstage/issues/20528
